src/partition/repart.c

```diff
@@ -2992,6 +2992,170 @@ static int context_wipe_and_discard(Context *context, bool from_scratch) {
         return 0;
 }
 
+typedef struct {
+        LoopDevice *loop;
+        int fd;
+        char *path;
+        int whole_fd;
+} PartitionTarget;
```

Comment:
so hmm, i don't like this.

What about instead beefing up LoopDevice? So that if you specify some flag it will actually wrap a regular file instead of a block device?

Reply:
That seems easier said than done. Looking at LoopDevice, it doesn't seem exactly trivial to modify it to support not using a loop device at all. I'll leave this out of the followups PR since it seems like a bigger thing compared to all the other nits. I'd be happy to review a PR or try to address it in a separate PR if you give some more details on how you think this should work.